File: src/main/java/com/teamwizardry/librarianlib/api/gui/IGuiDrawable.java
package com.teamwizardry.librarianlib.api.gui;

import com.teamwizardry.librarianlib.math.Vec2;

/**
 * An object that can be drawn to a bookcomponents
 * @author Pierce Corcoran
 */
public interface IGuiDrawable {

	/**
	 * Draw this object to the screen.
	 * @param mousePos
	 * @param partialTicks
	 */
	void draw(Vec2 mousePos, float partialTicks);
	
}

File: src/main/java/com/teamwizardry/librarianlib/client/gui/IGuiDrawable.kt
package com.teamwizardry.librarianlib.client.gui

import com.teamwizardry.librarianlib.common.util.math.Vec2d

/**
 * An object that can be drawn to a bookcomponents
 * @author Pierce Corcoran
 */
interface IGuiDrawable {

    /**
     * Draw this object to the screen.
     * @param mousePos
     * *
     * @param partialTicks
     */
    fun draw(mousePos: Vec2d, partialTicks: Float)

}
